### ISSUE TITLE
Automated deployment to testnet

### ISSUE THREAD

**Issue by birchmd** [aurora-is-near/aurora-engine#90]
We want a GitHub action which triggers on merge to `develop` which will deploy the new version of the contract to `develop.aurora` on testnet.

* Need to figure out the security of this (i.e. we do not want it to be possible for anyone to deploy to testnet, only the GitHub action)
* We will need more strict CI to ensure we don't accidentally break the testnet deployment. (See #89)

Anything else I've missed? Comments welcome.

**Comment by artob** [comment]
Sounds about right. Is it possible for CI to have access to less than a full access key, or is a full access key always required for code deployment?

**Comment by joshuajbouw** [comment]
Yeah, it makes sense to make. I know you can tailor Github actions to be quite amazing about triggering other actions with a key. Might it be worth it?

**Comment by birchmd** [comment]
> Is it possible for CI to have access to less than a full access key

I think this might be possible because of the way our upgrades work, but I'll have to test it.

Even if this is the case though, I think we would want to keep the key a secret to prevent others from pushing their own code to testnet. I'm not sure how to give a secret to an Action, but I'll look into it.

**Comment by birchmd** [comment]
Looks like this is how we will want to get the testnet key into a GitHub Action: https://docs.github.com/en/actions/reference/encrypted-secrets

@artob You will need to be the one to do this, or you will need to grant me `admin` access.
